kaldb/src/main/java/com/slack/kaldb/metadata/zookeeper/ZooKeeperCachedMetadataStoreImpl.java

```diff
@@ -254,7 +221,30 @@ public boolean isStarted() {
     return state.get().equals(State.STARTED);
   }
 
-  private void initialized() {
+  @VisibleForTesting
+  public boolean isStopped() {
+    return state.get().equals(State.STOPPED);
+  }
+
+  private void mayBeNotify() {
+    if (initializedLatch.getCount() == 0) {
```

Comment:
Just looked at this code to use AtomicBoolean and an initializedLatch is better since we `await` in `start` method. In this case, while not ideal, using `getCount` is ok since we only wait till the cache is initialized and it's always true after that. 

Also, this is existing code, so resolving it for now. If needed we can do this in a separate diff.

Reply:
Let's add a todo that this should be fixed. 

As it exists now there is a race condition where we can miss firing the `cacheChanged` event. Since the `countDown()` is being executed in a different thread, `getCount()` can return a stale value which would result in failing to notify listeners.